buildci.sh

```diff
@@ -62,11 +62,45 @@ if [[ ! -z "$CHANGED_DIR" ]] && [[ -e "bundles/$CHANGED_DIR" ]]; then
     # add the postfix to make sure we actually find the correct itest
     if [[ -e "itests/$CHANGED_DIR.tests" ]]; then
         echo "Single addon pull request: Building itest $CHANGED_DIR"
-        cd "itests/$CHANGED_DIR.tests"
+        # build the core index project so the OSGI index.xml file is generated in the target directory
+        # needed for resolving the OSGI bundle wiring later
+        cd "bom/openhab-core-index"
         mvn clean install -B 2>&1 |
-	      stdbuf -o0 grep -vE "Download(ed|ing) from [a-z.]+: https:" | # Filter out Download(s)
-	      stdbuf -o0 grep -v "target/code-analysis" | # filter out some debug code from reporting utility
-	      tee -a ${CDIR}/.build.log
+          stdbuf -o0 grep -vE "Download(ed|ing) from [a-z.]+: https:" | # Filter out Download(s)
+          stdbuf -o0 grep -v "target/code-analysis" | # filter out some debug code from reporting utility
+          tee -a ${CDIR}/.build.log
+        if [[ $? -ne 0 ]]; then
+            exit 1
+        fi
+         
+        # build the runtime index project so the OSGI index.xml file is generated in the target directory
+        # needed for resolving the OSGI bundle wiring later
+        cd "../runtime-index"
+        mvn clean install -B 2>&1 |
```

Comment:
Can you extracting the build commands to a function instead of duplicating them 4x?

Reply:
Or even better use a project list so they are all build in the same reactor using:

`-pl :org.openhab.addons.bom.openhab-core-index,:org.openhab.addons.bom.runtime-index,:org.openhab.addons.bom.test-index`

You can probably also add `:$CHANGED_DIR.tests:` to that list

Reply:
Good point, the -pl flag is a good idea, will switch to that.

Reply:
I applied the recommended changes, the script now uses the mvn -pl flag.